packages/bigtable/test/cluster.js

```diff
@@ -0,0 +1,274 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+var assert = require('assert');
+var format = require('string-format-obj');
+var proxyquire = require('proxyquire');
+var util = require('util');
+
+var GrpcServiceObject = require('@google-cloud/common').GrpcServiceObject;
+
+function FakeGrpcServiceObject() {
+  this.calledWith_ = arguments;
+  GrpcServiceObject.apply(this, arguments);
+}
+
+util.inherits(FakeGrpcServiceObject, GrpcServiceObject);
+
+describe('Bigtable/Cluster', function() {
+  var CLUSTER_NAME = 'my-cluster';
+  var PROJECT_ID = 'grape-spaceship-123';
+
+  var INSTANCE = {
+    id: 'projects/p/instances/i',
+    parent: { projectId: PROJECT_ID }
+  };
+
+  var CLUSTER_ID = format('{instance}/clusters/{cluster}', {
+    instance: INSTANCE.id,
+    cluster: CLUSTER_NAME
+  });
+
+  var Cluster;
+  var cluster;
+
+  before(function() {
+    Cluster = proxyquire('../src/cluster.js', {
+      '@google-cloud/common': {
+        GrpcServiceObject: FakeGrpcServiceObject
+      }
+    });
+  });
+
+  beforeEach(function() {
+    cluster = new Cluster(INSTANCE, CLUSTER_NAME);
+  });
+
+  describe('instantiation', function() {
+    it('should inherit from GrpcServiceObject', function() {
+      assert(cluster instanceof FakeGrpcServiceObject);
+
+      var config = cluster.calledWith_[0];
+
+      assert.strictEqual(config.parent, INSTANCE);
+      assert.strictEqual(config.id, CLUSTER_ID);
+
+      assert.deepEqual(config.methods, {
+        create: true,
+        delete: {
+          protoOpts: {
+            service: 'BigtableInstanceAdmin',
+            method: 'deleteCluster'
+          },
+          reqOpts: {
+            name: CLUSTER_ID
+          }
+        },
+        exists: true,
+        get: true,
+        getMetadata: {
+          protoOpts: {
+            service: 'BigtableInstanceAdmin',
+            method: 'getCluster'
+          },
+          reqOpts: {
+            name: CLUSTER_ID
+          }
+        }
+      });
+    });
+
+    it('should Instance#createCluster to create the cluster', function(done) {
+      var fakeOptions = {};
+
+      INSTANCE.createCluster = function(name, options, callback) {
+        assert.strictEqual(name, CLUSTER_NAME);
+        assert.strictEqual(options, fakeOptions);
+        callback();
+      };
+
+      cluster.createMethod(null, fakeOptions, done);
+    });
+
+    it('should leave full cluster names unaltered', function() {
```

Comment:
Actually that is incorrect (kinda). In our `beforeEach` hook, we create a new Cluster before each test that uses a partial name. In our very first test we assert that the cluster has an id and that is the full name.

Do you want an additional test for the sake of being explicit?